src/smaug/errors.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+export function TokenError(message) {
+  this.name = 'TokenError';
+  this.message = message || 'Generic TokenError';
+  this.stack = (new Error()).stack;
+  this.httpStatusCode = 400;
+  this.httpError = 'generic_token_error';
+}
+TokenError.prototype = Object.create(Error.prototype);
+TokenError.prototype.constructor = TokenError;
```

Comment:
Makes sense, - when you override the prototype, the constructor is no longer set, and therefore you have to set it manually. My mistake :)